src/plugins/application_config/server/plugin.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Observable } from 'rxjs';
+import { first } from 'rxjs/operators';
+
+import {
+  PluginInitializerContext,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  Logger,
+  IScopedClusterClient,
+  SharedGlobalConfig,
+} from '../../../core/server';
+
+import {
+  ApplicationConfigPluginSetup,
+  ApplicationConfigPluginStart,
+  ConfigurationClient,
+} from './types';
+import { defineRoutes } from './routes';
+import { OpenSearchConfigurationClient } from './opensearch_config_client';
+
+export class ApplicationConfigPlugin
+  implements Plugin<ApplicationConfigPluginSetup, ApplicationConfigPluginStart> {
+  private readonly logger: Logger;
+  private readonly config$: Observable<SharedGlobalConfig>;
+
+  private configurationClient: ConfigurationClient;
+  private configurationIndexName: string;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
+    this.config$ = initializerContext.config.legacy.globalConfig$;
+    this.configurationIndexName = '';
+  }
+
+  private setConfigurationClient(inputConfigurationClient: ConfigurationClient) {
+    this.configurationClient = inputConfigurationClient;
+  }
+
+  private getConfigurationClient(inputOpenSearchClient: IScopedClusterClient) {
+    if (this.configurationClient) {
+      return this.configurationClient;
+    }
+
+    const openSearchConfigurationClient = new OpenSearchConfigurationClient(
+      inputOpenSearchClient,
+      this.configurationIndexName,
+      this.logger
+    );
+
+    this.setConfigurationClient(openSearchConfigurationClient);
+    return this.configurationClient;
+  }
+
+  public async setup(core: CoreSetup) {
+    this.logger.debug('applicationConfig: Setup');
```

Comment:
Do we need this log?

Reply:
Automatically created by plugin generator. Just removed.